File: 250504/XOR 결과 최대 만들기/max-of-xor.cpp
#include <iostream>
#include <cmath>
#include <vector>
using namespace std;

int n, m;
int A[20];
vector<int> v;
int ans = -1;
void print() {
    for(int a : v) cout << a << ' ';
    cout<<'\n';
}
int calc() {
    int result = 0;
    for (int i = 0; i < v.size(); i++) {
        result = result ^ v[i];
    }
    return result;
}

void func(int idx, int cnt) {
    if(cnt == m) {
        ans = max(ans, calc());
        return;
    }
    if(idx + 1 < n) {
        v.push_back(A[idx]);
        func(idx + 1, cnt + 1);
        v.pop_back();

        func(idx + 1, cnt);
    }

}
int main() {
    cin >> n >> m;

    for (int i = 0; i < n; i++) {
        cin >> A[i];
    }

    func(0, 0);
    cout << ans;
    return 0;
}

File: 250810/XOR 결과 최대 만들기/max-of-xor.py
n, m = map(int, input().split())
A = list(map(int, input().split()))
selected = []
ans = 0
def calc():
    result = 0
    for num in selected:
        result = result ^ num
    return result

def func(idx, cnt):
    global ans
    if cnt == m:
        ans = max(ans, calc())
        return
    elif idx > n:
        return
    else:
        selected.append(idx)
        func(idx + 1, cnt + 1)
        selected.pop()
        func(idx + 1, cnt)

func(1, 0)
print(ans)